### ISSUE TITLE
DDP.no_sync() function + torch.cuda.synchronize(device=rank)

### ISSUE THREAD

**Issue by densechen** [pytorch/pytorch#53443]
I find the using example in doc:
```
>>> ddp = torch.nn.parallel.DistributedDataParallel(model, pg)
>>> with ddp.no_sync():
>>>   for input in inputs:
>>>     ddp(input).backward()  # no synchronization, accumulate grads
>>> ddp(another_input).backward()  # synchronize grads
```

In this example, we have to use a new data, i.e. `another_input`, to synchronize grads.
Can this operation be done via ```torch.cuda.synchronize(device=rank)```?

**Comment by pritamdamania87** [comment]
> Can this operation be done via torch.cuda.synchronize(device=rank)?

No, that would only synchronize the local cuda device. For DDP, each process is accumulating grads locally in the `no_sync` mode and as a result have different grads. Once, you run without no_sync, it will synchronize (allreduce) grads across all processes to ensure all processes have the same grads.

**Comment by densechen** [comment]
@pritamdamania87 Thanks for your reply! By the way, does there exist other way to synchronize grads instead of ```ddp(another_input).backward()```?
